texlive: 2019 -> 2020

veprbl opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

Provide minor fixes for some items in `texlive.bin`. Update to the latest texlive snapshot.
Closes: #91994

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
```
nix path-info -h -S /nix/store/vb3l4zhkigrniqh400vl96qly8kpbhd1-texlive-combined-full-2020 -S /nix/store/0rlzarjqrplckcfdydk22na48dfj7zk9-texlive-combined-full-2019
/nix/store/0rlzarjqrplckcfdydk22na48dfj7zk9-texlive-combined-full-2019     3.4G
/nix/store/vb3l4zhkigrniqh400vl96qly8kpbhd1-texlive-combined-full-2020     3.8G
```
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

veprbl committed:
texlive.bin.core: remove outdated options

veprbl committed:
texlive.bin.core: avoid building lua{53,jit} libraries

veprbl committed:
texlive.bin.mflua: init

veprbl committed:
texlive.combined.scheme-infraonly: enable building on Hydra

This used to be broken, but recently was fixed.

veprbl committed:
texlive: 2019 -> 2020

Also change default package version from texlive year to a
revision. The problem with using texlive year is that it will change
on each of our update even if the package itself was not updated. The
texlive SVN revision number should not change unless the package is
updated. This will allow to share more store paths between different
texlive versions.

gebner commented:
Builds and works for me (tested on a few latex files).

gebner commented:
This PR is still marked as draft, but pretty much LGTM from my side.  Is there anything that you still want to do?

veprbl committed:
texlive.bin: provide luajittex and mfluajit binaries

The ppc64 condition is based on previous reports and was not tested on
actual hardware.

veprbl commented:
@gebner Just pushed a new changeset. Main difference is a change of default versioning scheme for unversioned packages, it now uses SVN revision number instead of the texlive year. There are also minor tweaks for luajit (cc @CrystalGamma for ppc64 changes).

We usually had major texlive updates sit for a while before getting merged, but we could try merging sooner this time.

veprbl commented:
@GrahamcOfBorg build tests.texlive texlive.combined.scheme-full

gebner commented:
> We usually had major texlive updates sit for a while before getting merged, but we could try merging sooner this time.

I know, in 2017 I merged the texlive update after it had been sitting around for over a month with no activity, and we only got bug reports after it was merged.

We don't gain much by letting this PR sit longer than necessary.  If it builds, then I think we should merge it.

gebner commented:
I ran into one regression, the new biber 2.15 doesn't like authors such as `Jean{-}Marc`.  This is already fixed upstream (but not released yet): https://github.com/plk/biber/issues/329

dtzWill commented on pkgs/tools/typesetting/tex/texlive/bin.nix:
```diff
@@ -14,78 +14,26 @@
-      })
-    ];
-
     # remove when removing synctex-missing-header.patch
```
Think this should've been removed before, per:comment?

veprbl replied:
@dtzWill Nice catch, this should have been addressed back in update to 2019. Opened #101016. Thanks!
